analytics.ts

```diff
@@ -0,0 +1,121 @@
+import * as Settings from '@taqueria/protocol/Settings';
+import { chain, map, resolve } from 'fluture';
+import { pipe } from 'https://deno.land/x/fun@v1.0.0/fns.ts';
+import { getMachineId } from 'https://deno.land/x/machine_id@v0.3.0/mod.ts';
+import type { DenoArgs } from './taqueria-types.ts';
+import * as utils from './taqueria-utils/taqueria-utils.ts';
+
+const {
+	doesPathExist,
+	readJsonFile,
+	writeJsonFile,
+	eager,
+} = utils.inject({
+	stdout: Deno.stdout,
+	stderr: Deno.stderr,
+});
+
+const consentPrompt =
+	'Allow Taqueria anonymously report usage statistics, in accordance with the privacy policy to help make Taqueria better? [y/yes] or [n/no]';
+const optInConfirmationPrompt =
+	'Allow Taqueria anonymously report usage statistics, in accordance with the privacy policy to help make Taqueria better? [y/yes] or [n/no]';
+const optOutConfirmationPrompt = 'Are you sure you want to turn off usage statistic reporting? [y/yes] or [n/no]';
+export const settingsFilePath = Deno.env.get('HOME') + '/.taq-settings.json';
+export const OPT_IN = 'opt_in';
+export const OPT_OUT = 'opt_out';
+
+export const optInAnalyticsFirstTime = () => createSettingsFileWithConsent(OPT_IN);
+
+export const optOutAnalyticsFirstTime = () => createSettingsFileWithConsent(OPT_OUT);
+
+const createSettingsFileWithConsent = (option: 'opt_in' | 'opt_out') =>
+	pipe(
+		Settings.make({ consent: option }),
+		chain(writeJsonFile(settingsFilePath)),
+	);
+
+export const optInAnalytics = () => writeConsentValueToSettings(OPT_IN);
+
+export const optOutAnalytics = () => writeConsentValueToSettings(OPT_OUT);
+
+const writeConsentValueToSettings = (option: 'opt_in' | 'opt_out') => {
+	const input = prompt(option === OPT_IN ? optInConfirmationPrompt : optOutConfirmationPrompt);
+	if (input && /^y(es)?$/i.test(input)) {
+		return pipe(
+			readJsonFile<Settings.t>(settingsFilePath),
+			map((settingsContent: Settings.t) => {
+				settingsContent.consent = option;
+				return settingsContent;
+			}),
+			chain(writeJsonFile(settingsFilePath)),
+			map(() => true),
+		);
+	} else {
+		return resolve(false);
+	}
+};
+
+const promptForConsent = async () => {
+	const input = prompt(consentPrompt);
+	if (input && /^y(es)?$/i.test(input)) {
```

Comment:
This considers any input other than y or yes as a no.
Do we want to repeat the prompt instead, if `y`/`n`/`yes`/`no` is not given?

Reply:
Yes we need to have an exact confirmation from the user (either yes/y or no/n). A non matching response should produce an error output and re-prompt as Ali mentioned

Reply:
What do you mean presses ` `? A space bar?

And what other keys will map to "yes" aside from capital `Y`? What about small `y`?